aiida/orm/nodes/data/array/bands/electronic.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+"""Data plugin to represent an electronic band structure."""
+from typing import Optional
+
+from aiida.common.exceptions import ModificationNotAllowed
+from aiida.common.lang import type_check
+from .bands import BandsData
+
+__all__ = ('ElectronicBandsData',)
+
+
+class ElectronicBandsData(BandsData):
+    """Data plugin to represent an electronic band structure.
+
+    It subclasses the ``BandsData`` plugin and adds the ``fermi_energy`` attribute.
+    """
+
+    KEY_FERMI_ENERGY = 'fermi_energy'
+
+    def __init__(self, **kwargs):
```

Comment:
I must admit I do not like this constructor using `**kwargs` from a UX point of view. ^^ Basically it's impossible to tell for a user that checks the docstring how to use it. What's the advantage of this, and does it outweigh the benefit of the user simply seeing the input arguments and having them documented in the docstring?

Reply:
I have changed the interface. However, it does bring other potential use problems with it. See the commit message for an explanation. Using the `/` marker to make them positional only "solves" the problem somewhat.

Reply:
Hmmm, the `/` positional argument only marker is only supported for Python 3.8 and up. Not sure if there is a backport solution for this, but I doubt it since it concerns the basic syntax. @chrisjsewell do you happen to know?